Surface_mesh_parameterization/include/CGAL/Surface_mesh_parameterization/MVC_post_processor_3.h

```diff
@@ -418,27 +386,25 @@ class MVC_post_processor_3
     // The other parts of A(i,j) and A(i,k) will be added when this function
     // is called from the neighboring faces of F_ijk that share the vertex i
 
-    // Compute: - tan(alpha / 2)
-    NT w_i_base = -1.0 * compute_w_ij_mvc(pi, pj, pk);
-
     // @fixme unefficient: lengths are computed (and inversed!) twice per edge
 
+    // Set w_i_base: - tan(alpha / 2)
+    const Point_2& p = pk;
+    const Point_2& q = pi;
+    const Point_2& r = pj;
+    const CGAL::Weights::internal::
+      Tangent_weight_wrapper<NT> tangent_weight(p, q, r);
+
     // Set w_ij in matrix
-    Vector_2 edge_ij = pi - pj;
-    double len_ij = std::sqrt(edge_ij * edge_ij);
-    CGAL_assertion(len_ij != 0.0); // two points are identical!
-    NT w_ij = w_i_base / len_ij;
+    const NT w_ij = tangent_weight.get_w_r();
     A.add_coef(i, j, w_ij);
 
     // Set w_ik in matrix
-    Vector_2 edge_ik = pi - pk;
-    double len_ik = std::sqrt(edge_ik * edge_ik);
-    CGAL_assertion(len_ik != 0.0); // two points are identical!
-    NT w_ik = w_i_base / len_ik;
+    const NT w_ik = tangent_weight.get_w_p();
```

Comment:
In case weights work on number types, how do you deal with `sqrt`?

Reply:
@MaelRL and @sloriot is that ok?

Reply:
https://github.com/CGAL/cgal/blob/cac04ed1496a6c17c8c4b16dfa09c20999204595/Weights/include/CGAL/Weights/internal/utils.h#L64

Reply:
@MaelRL ping

Reply:
We are moving here like at other places towards having some kind of `CGAL::approximate_sqrt` everywhere. I feel like this needs a proper discussion regardless of this PR:
- The function `CGAL::approximate_sqrt` does the job, but its name is bad because sometimes it's not approximate;
- I would put a macro like "CGAL_DISABLE_APPROXIMATE_SQRT" that would only allow exact sqrt in that generic call, to be able to easily see if you are calling a sqrt in a given piece of code.

Reply:
@MaelRL Should I change something in this PR?

Reply:
No, just use `CGAL::approximate_sqrt`.

Reply:
I create an issue about that: #5807.